gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -425,6 +434,47 @@ public static List<Service> parseService(
         Transport.GRPC);
   }
 
+  private static Optional<List<String>> getInclusionMethodListFromServiceYaml(
+      Optional<com.google.api.Service> serviceYamlProtoOpt, String protoPackage) {
+    if (!serviceYamlProtoOpt.isPresent()
+        || serviceYamlProtoOpt.get().getPublishing().getLibrarySettingsCount() == 0) {
+      return Optional.empty();
+    }
+    List<ClientLibrarySettings> librarySettingsList =
+        serviceYamlProtoOpt.get().getPublishing().getLibrarySettingsList();
+    // verify library_settings.version matches with proto package name
+    // This should be validated in upstream. Give warnings if not match.
+    if (!librarySettingsList.get(0).getVersion().isEmpty()
+        && !protoPackage.equals(librarySettingsList.get(0).getVersion())) {
+      LOGGER.warning(
+          "Service yaml config may be misconfigured. "
+              + "Version in publishing.library_settings does not match proto package."
+              + "Disregarding selective generation settings.");
+      return Optional.empty();
+    }
+    List<String> includeMethodsList =
+        librarySettingsList
+            .get(0)
+            .getJavaSettings()
+            .getCommon()
+            .getSelectiveGapicGeneration()
+            .getMethodsList();
+    return Optional.of(includeMethodsList);
+  }
+
+  private static boolean shouldIncludeMethodInGeneration(
```

Comment:
Can we combine this method with `getInclusionMethodListFromServiceYaml`? To something like `shouldIncludeMethod(String methodName, com.google.api.Service serviceYaml)`? This also removes the need of returning `Optional<List<String>>`, which does not seem to provide much benefit. Was it for performance concerns?

Reply:
As discussed off-line, combining the two methods.